### PR TITLE
fix: same dialog design for join event and join video

### LINKED ISSUE #6364: Virtual Events Sidebar: Only provide option to view sidebar for attendees with tickets
When an anonymous users accesses a "Join Video" link they will get a pop up to login or to get a ticket. If the user presses "Cancel" the event page and the side panel  will show up even for anonymous users. Users using the "Join Event" link see another message and do not have the option "Cancel".

Expected is: 
* That only the event page would be visible without the side panel. 
* The pop up message, buttons and design should be same for both "Join Event" and "Join Video" link.

Example https://eventyay.com/e/f4fc71b8?side_panel=true&video_dialog=true

Screenshot from 2021-01-14 14-35-43

Screenshot from 2021-01-14 14-36-00

### PR DESCRIPTION
**Pull request by ranjitkshah**
Fixes #6364 

#### Short description of what this resolves:
- [ ] That only the event page would be visible without the side panel.
- [x] The pop up message, buttons and design should be same for both "Join Event" and "Join Video" link.

#### Screenshot:
https://www.loom.com/share/dffc067050e54998b4b17d38589717c7

-
-
-

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/components/modals/simple-modal.js [hunk 1]
```diff
@@ -9,4 +9,10 @@ export default class SimpleModal extends ModalBase {
     this.close();
     this.action();
   }
+
+  @action
+  performCancel() {
+    this.close();
+    this.router.transitionTo('public', { queryParams: { video_dialog: null } });
```
Comment (iamareebjamal) [inline app/components/modals/simple-modal.js:16]: Seriously? This component is not just there for video dialog, it is a generic component

File: app/components/modals/simple-modal.hbs [hunk 2]
```diff
@@ -13,7 +13,14 @@
       {{t-var (or @actionName 'OK')}}
     </button>
   {{/if}}
+  {{#if @cancel}}
+    <button type="button" class="ui button" {{action 'close'}} {{action @cancel}}>
+    {{t-var (or @cancelName 'Cancel')}}
+  </button>
+  {{/if}}
```
Comment (iamareebjamal) [inline app/components/modals/simple-modal.hbs:20]: Duplication

Comment (ranjitkshah) [reply app/components/modals/simple-modal.hbs:20]: please suggest better approach

Comment (iamareebjamal) [reply app/components/modals/simple-modal.hbs:20]: or

File: app/components/modals/simple-modal.hbs [hunk 3]
```diff
@@ -13,7 +13,7 @@
       {{t-var (or @actionName 'OK')}}
     </button>
   {{/if}}
-  <button type="button" class="ui button" {{action 'close'}}>
+    <button type="button" class="ui button" {{action 'close'}} {{action @cancel}}>
```
Comment (ranjitkshah) [inline app/components/modals/simple-modal.hbs:16]: @iamareebjamal confuse here, how to use OR condition between actions please suggest

**Comment by afzal442** [comment]
plz also remove close icon from both. not required as cancel button does the job.

**Comment by iamareebjamal** [comment]
No

**Comment by ranjitkshah** [comment]
@iamareebjamal  pls check now

**Review by mariobehling — APPROVED** [review]
Works as expected for me. Thanks!

File: app/components/modals/simple-modal.hbs [hunk 4]
```diff
@@ -13,7 +13,7 @@
       {{t-var (or @actionName 'OK')}}
     </button>
   {{/if}}
-  <button type="button" class="ui button" {{action 'close'}}>
+    <button type="button" class="ui button" {{action this.performCancel}}>
```
Comment (iamareebjamal) [inline app/components/modals/simple-modal.hbs:16]:
```suggestion
  <button type="button" class="ui button" {{action this.performCancel}}>
```

File: app/controllers/public.js [hunk 5]
```diff
@@ -49,4 +49,9 @@ export default class PublicController extends Controller {
       this.transitionToRoute('login');
     }
   }
+
+  @action
+  cancel() {
```
Comment (iamareebjamal) [inline app/controllers/public.js:54]: Rename to closeVideoDialog

**Pull request merged**